source/plugins/developing.rst

```diff
@@ -21,77 +21,10 @@ Writing a simple QIIME 2 plugin should be a straightforward process. For example
-
-For example, to template a plugin that is compatible with the 2017.2 QIIME 2 release, choose a 2017.2 release tag from the `template repository releases <https://github.com/qiime2/cookiecutter-plugin-template/releases>`_. The latest patch number within a release is recommended.
-
-.. command-block::
-   :no-exec:
-
-   cookiecutter -c 2017.2.0 https://github.com/qiime2/cookiecutter-plugin-template
-
-This initializes a plugin package in your current working directory. Use ``-o``/``--output-dir`` to create a plugin package in a location other than the current working directory.
-
-After running the above command, you will be prompted to enter several pieces of information about your plugin, including the plugin name, description, author details, etc. **Defaults are provided as examples only**; you must provide the relevant information about your plugin. The prompts only gather basic information about your plugin so a functioning package can be initialized. There are other pieces of the plugin that can be manually configured (detailed below).
-
-Assuming all default values were used in response to the above command's prompts, you should see that a ``q2-my-plugin`` directory was created in your current working directory. This directory contains the initialized plugin package based on the information you provided.
-
-The plugin includes some example functionality that you can try out (for example, using ``q2cli``). If you intend to use the plugin's example functionality, install `q2-dummy-types`_, which provides the semantic types used by the plugin:
-
-.. command-block::
-   :no-exec:
-
-   conda install --override-channels -c qiime2 -c defaults q2-dummy-types
-
-Next, navigate to the plugin directory that was created and install the plugin in development mode:
-
-.. command-block::
-   :no-exec:
-
-   cd q2-my-plugin
-   pip install -e .
-
-To see that the plugin is discoverable by QIIME, run:
-
-.. command-block::
-   :no-exec:
-
-   qiime
-
-You should see ``my-plugin`` listed as one of the available commands. To see the available plugin commands:
-
-.. command-block::
-   :no-exec:
-
-   qiime my-plugin --help
-
-Once you are done exploring the plugin's example functionality, update it with your own. The relevant sections of the code that need to change are commented.
-
-.. note::
-
-   If you are testing your plugin with ``q2cli`` (i.e. the ``qiime`` command) while you are developing it, you'll need to run ``qiime dev refresh-cache`` to see the latest changes to your plugin reflected in the CLI. You'll need to run this command anytime you modify your plugin's interface (e.g. add/rename/remove a command or its inputs/parameters/outputs).
```

Comment:
We'll want this note and everything below it I think. We may need to just turn this section into something that describes how Plugins are found by the framework. And then it would make sense to mention that q2cli will cache this information, so you can work around that with these steps.